### ISSUE TITLE
'Delete role' feature

### ISSUE THREAD

**Issue by olemp** [Puzzlepart/did#692]
- [x] I have searched for existing issues to avoid creating duplicates

### TLDR
We should be able to delete a role.

### Description
At the moment we cannot delete roles. We need to consider implementing this, at least as long as the role is not in use.

The default roles `User`, `Admin` and `Owner` should be read only.

I create a new role, and assign it to a couple of users, and then want to delete the role.

* Make the role undeletable while it is in use - like content types in SharePoint.
  * Do a query for users with [role to delete], and if there are any, list them out and alert the user
`"This role cannot be deleted because it is in use by 'user1' and 'user2'`
* V2: Feature for converting all users with [role to delete] to [another role chosen from dropdown].

See #691
